otio_openassetio/operations/openassetio_media_linker.py

```diff
@@ -44,32 +44,45 @@ def link_media_reference(in_clip, media_linker_argument_map):
     # over time, however this function isn't, so we track a shared
     # session based on the arguments to the media linker.
     session_state = _sessionState(media_linker_argument_map)
-    manager = session_state.session.currentManager()
+    manager = session_state.manager
 
-    if not extract_one(manager.isEntityReference([mr.target_url])):
+    if not manager.isEntityReferenceString(mr.target_url):
         return
 
+    entity_reference = manager.createEntityReference(mr.target_url)
+
     # Update the locale with more information about this call
     # This is perhaps more illustrative than useful at this point.
     context = session_state.context
     ClipTrait(context.locale).setName(in_clip.name)
 
+    # Upon a successful resolve, use the concrete accessors of the trait
+    # as a view on the resolved data to ensure we access the correct
+    # keys/values.
+    def successful_resolve_callback(_, entity_data):
+        locatable = LocatableContentTrait(entity_data)
+        if locatable.isValid():
```

Comment:
[fix] Adding this check has removed the exception if the trait wasn't set. We need that exception so the user knows something is wrong.

(PS. If we were keeping the check, def worth re-using `locatable` for the value retrieval.)

Reply:
[49c5e57](https://github.com/OpenAssetIO/otio-openassetio/pull/17/commits/49c5e5740f836b4841ffe01c17618374a0c1bfdb)

Reply:
[8dfe884](https://github.com/OpenAssetIO/otio-openassetio/pull/17/commits/8dfe884c4112891e40418adeb9f40cb205929b9b)